metastore/common/errors.go

```diff
@@ -77,4 +77,6 @@ var (
 	ErrHLLColumnDoesNotAllowDefaultValue = errors.New("hll column does not allow default value")
 	ErrInvalidTableBatchSize             = errors.New("Table batch size should be larger than zero")
 	ErrInvalidPrimaryKeyBucketSize       = errors.New("Table primary key bucket size should be larger than zero")
+	ErrInvalidPrimaryKeyDataType         = errors.New("Array data type can not used as primary key")
```

Comment:
error msg could be more generic, to match error type - we could have other types that are not eligible as PK in the future.